src/client/datascience/jupyter/kernels/kernel.ts

```diff
@@ -659,41 +659,14 @@ export class Kernel implements IKernel {
             };
             this.disposables.push(notebook.session.onSessionStatusChanged(statusChangeHandler));
         }
-
         if (isPythonKernelConnection(this.kernelConnectionMetadata)) {
             // So that we don't have problems with ipywidgets, always register the default ipywidgets comm target.
             // Restart sessions and retries might make this hard to do correctly otherwise.
             notebook.session.registerCommTarget(Identifiers.DefaultCommTarget, noop);
 
-            if (isLocalConnection(this.kernelConnectionMetadata)) {
```

Comment:
This refactor (executing all the code in once cell) lopped off about 3 seconds for me for kernel startup.